src/Middleware/Diagnostics/src/DeveloperExceptionPage/IDeveloperPageExceptionFilter.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Diagnostics
+{
+    public interface IDeveloperPageExceptionFilter
+    {
+        Task HandleExceptionAsync(HttpContext context, Exception exception, Func<HttpContext, Exception, Task> next);
```

Comment:
An error context object would be preferred. It simplifies the signature and gives you future proofing.
`Task HandleExceptionAsync(ErrorContext context, Func<ErrorContext, Task> next);`

ErrorContext:
- HttpContext
- Exception

Reply:
Sure I'll add the an ErrorContext. I need to say it though, these types should really be structs (generally). It's such a waste to allocate an object that wraps a bunch of other objects. This is an error path and is only used in development so I care less.

Reply:
Also, mutable or immutable? I'd like to make it immutable but then it brings up the question on adding another constructor every time we want to add a field.

Reply:
Mutable, it's easier to test and you don't need any constructors.

Reply:
Made it immutable. Its cleaner as both of those parameters are required to be non null.